src/uniswap.ts

```diff
@@ -409,27 +465,46 @@ export const getReportOptions = (
   };
 };
 
-export const configForNetwork = (network: string): Config => {
+export const configForNetworkAndPair = (
+  network: string,
+  pair: Pair
+): Config => {
   const lcNetwork = network.toLowerCase();
-  const config = CONFIGURATIONS[lcNetwork];
 
-  if (lcNetwork === "mainnet") {
+  const unsupported: { [network: string]: boolean } = {
+    mainnet: true,
+    rinkeby: true,
+  };
+  if (unsupported[lcNetwork]) {
```

Comment:
With type changes above this `if` may be removed because the type system would ensure that value is correct

Reply:
This was somewhat of a special case.
Explaining why `mainnet` and `rinkeby` specifically are not supported at the moment.
But, I guess, it is fine to just remove it.
As with proper types, I would actually need to pass unsupported values deeper into the application logic.